### ISSUE TITLE
Spirit of Eskhandar triggers Pyroguard Emberseer (UBRS) before he's is meant to be engaged

### ISSUE THREAD

**Issue by TQPS** [TrinityCore/TrinityCore#27727]
### Description

He's a weird one for you ...

If you have 2 Eskhandar pieces equipped, you have a change to spawn the [Spirit of Eskhandar](https://classic.wowhead.com/item-set=261/spirit-of-eskhandar).

If it procs before you enter Pyroguard Emberseer's room, then he will immediately come down off his platform, attack Eskhandar until he's dead, not engage anything else, return to his platform and the encounter resets.

<details>

![image](https://user-images.githubusercontent.com/66294837/152603148-e51362b9-5021-452e-8019-efcc2a3e3d64.png)

</details>

If it procs while you are engaging the summoners, then he will come down off his platform, attack Eskhandar until he's dead, not engage anything else, return to his platform and the encounter resets. You then need to restart the encounter.

<details>

![image](https://user-images.githubusercontent.com/66294837/152603501-555f610e-83f7-4d1b-b592-467219d6ac41.png)

</details>

You pretty have to pray that the Spirit of Eskhandar does not proc anywhere prior or during the Pyroguard encounter.

* As a side-note, I have seen it proc in Stratholme, engage in combat with Skeletons, they kill it and then return to their patrols without engaging the rest of the group.

### Expected behaviour

The Spirit of Eskhandar spawn should not invoke an encounter with Pyroguard Emberseer.

### Steps to reproduce the problem

1. Have equipped or a bot equipped with 2 pieces of Eskhandar gear
Either
2. Have the items proc prior to entering the room
or
2. Have the items proc inside Pyroguard's room during summoner fight
3. Pyroguard Emberseer will come down, kill Eskhandar (after a number of hits) and then return to his spawn and the encounter resets - reguardless of the fight progress.

### Branch

3.3.5

### TC rev. hash/commit

TrinityCore rev. https://github.com/TrinityCore/TrinityCore/commit/a650bd631a4681d09e3ff66f79b73b99299565f3 2022-01-16 18:50:28 +0100 (3.3.5 branch) (Win64, RelWithDebInfo, Static)

### Operating system

Windows 7 x64

### Custom changes

Trinity Bots

**Comment by offl** [comment]
He has no UNIT_FLAG_IMMUNE_TO_NPC in 4xx \ classic sniffs

UNIT_FLAG_IMMUNE_TO_PC he uses prevents player's pets to enter with combat with creatures (hunter, warlock pets), but this creature is a guardian and maybe that part is not implemented

**Comment by Alastro-Kraken** [comment]
It would make sense that UNIT_FLAG_IMMUNE_TO_PC extends to anything that's even remotely player-related.

**Comment by offl** [comment]
>It would make sense that UNIT_FLAG_IMMUNE_TO_PC extends to anything that's even remotely player-related.

And looks like actually it works, I spawned Eskhandar and other guardians but Emberseer doesn't attack them

.go c id 9816
.cast 22649 / 12790

**Comment by TQPS** [comment]
Is this a bot issue then? If the items proc on a bot which spawns Eskhandar then I should raise this withTrickerer?

**Comment by offl** [comment]
I guess
You can try to check unit_flags of Eskhandar using .npc info
If it does not have UNIT_FLAG_PLAYER_CONTROLLED, then it's problem with bots

**Comment by Alastro-Kraken** [comment]
> Is this a bot issue then? If the items proc on a bot which spawns Eskhandar then I should raise this withTrickerer?

Correct me if I'm wrong, but those are NPCBots, and not Ike3's Playerbots, right?

Big difference between the two, because Playerbots are actual players (have a worldsession, loaded from characters db, just like real players), but NPCBots are mostly creatures, in which case their summons wouldn't be treated the same as player ones (think of a player summoning an Imp versus a demon NPC summoning an Imp).

**Comment by TQPS** [comment]
Fair enough, I'll close it and accept it as a known issue for Trickerer's Bots.

I'll replace an items on the 2 bots when something drops.

**Comment by offl** [comment]
Yeah, those are npcs, there may be more problems like this one

**Comment by TQPS** [comment]
For reference only ...

<details>

![image](https://user-images.githubusercontent.com/66294837/152619480-2270b9f6-b542-46ec-8c42-6fd6552e4a1c.png)

</details>
